### PR TITLE
GWW-183: surface volume area chart

### PR DESCRIPTION
**Pull request by lsarni**
Ticket: https://issuetracker.deltares.nl/browse/GWW-183

**Changes:**
- Surface Volume Area chart is shown for some single reservoirs (for example: 82673)
- If no volume area data is available show a message
- Include the Surface Volume Area data on the csv download

image

### CODE REVIEW

File: src/assets/theme/chart-theme.json [hunk 1]
```diff
@@ -20,6 +20,9 @@
             "color": "rgba(255,255,255,0.5)"
         }
     },
+    "grid": {
+        "containLabel": true
+    },
```
Comment (lsarni) [inline src/assets/theme/chart-theme.json:25]: Prevents the label from getting cropped like this 
image

File: src/repo/reservoirRepo.js [hunk 2]
```diff
@@ -53,30 +40,28 @@ const timeSeriesAxis = (yAxisName) => {
     ],
     yAxis: [
       {
-        name: `${yAxisName} (${valueUnit})`,
+        name: `${yAxisName} (${yAxisUnit})`,
         type: 'value',
+        nameLocation: 'middle',
+        nameGap: 50,
```
Comment (lsarni) [inline src/repo/reservoirRepo.js:46]: It rotates the label to appear parallel to the axis

**Review by SiggyF — APPROVED** [review]
Looks good!

File: src/repo/reservoirRepo.js [hunk 3]
```diff
@@ -1,42 +1,29 @@
...
-const valueUnit = 'km2'
+const unitConversion = (unit) => {
+  if (unit === 'm2') {
+    return 'km2'
+  } else if (unit === 'm3') {
+    return 'km3'
+  } else {
+    return unit
+  }
+}
```
Comment (zowievangeest) [inline src/repo/reservoirRepo.js:24]: Maybe we can combine these 2 in a helper function where we return a label and value so it makes more sense that those belong to each other.

**Review by zowievangeest — CHANGES_REQUESTED** [review]

File: src/repo/reservoirRepo.js (hunk 3 above)
Comment (SiggyF) [reply src/repo/reservoirRepo.js:24]: There are some concepts of "beautiful" numbers. In d3 for example they have the 's' SI unit notation:
https://github.com/d3/d3-format/tree/v3.1.0

**Pull request merged**